File: Backend/Events/Converters/Parties/DigievolutionSlotConverter.cs
using Backend.Domain.Models.Parties.Digimons;
using Backend.Events.DTO.Party.Digimon;

namespace Backend.Events.Converters.Parties;

public static class DigievolutionSlotConverter
{
    public static DigievolutionSlotDTO ToDTO(DigievolutionSlot slot) => new()
    {
        Index = slot.Index,
        DigievolutionId = slot.DigievolutionId,
        Digievolution = slot.Digievolution != null ? DigievolutionConverter.ToDTO(slot.Digievolution) : null
    };
}

File: Frontend/src/events/converters/parties/digimons/digievolution-slot.converter.ts
import type { DeepRequired } from "@/events/dto/deep-required";
import type { DigievolutionSlotDTO } from "@/events/dto/parties/digimons/digievolution-slot.dto";
import type { DigievolutionSlot } from "@/models";
import { DigievolutionConverter } from "./digievolution.converter";

export class DigievolutionSlotConverter {
  public static convert(digievolutionSlotDto: DeepRequired<DigievolutionSlotDTO>): DigievolutionSlot {
    return {
      index: digievolutionSlotDto.index,
      digievolutionId: digievolutionSlotDto.digievolutionId,
      digievolution: digievolutionSlotDto.digievolution
        ? DigievolutionConverter.convert(digievolutionSlotDto.digievolution)
        : null,
    };
  }
}
